File: src/main/java/net/rentalhost/plugins/php/hammer/services/StringService.java
package net.rentalhost.plugins.php.hammer.services;

import org.apache.commons.lang.StringUtils;

import java.util.Arrays;
import java.util.stream.Stream;

public class StringService {
    public static Stream<String> listNonNullableTypes(final String types) {
        return Arrays.stream(StringUtils.split(types, "|"))
                     .filter(s -> !s.equals("null") && !s.equals("\\null"));
    }
}

File: src/main/kotlin/net/rentalhost/plugins/php/hammer/services/StringService.kt
package net.rentalhost.plugins.php.hammer.services

import com.jetbrains.php.lang.psi.elements.StringLiteralExpression

object StringService {
    fun addQuotes(string: String, isComplex: Boolean, addSlashes: Boolean = true): String =
        if (isComplex) {
            if (addSlashes) "\"${string.replace("\\", "\\\\")}\""
            else "\"${string}\""
        } else "'${string}'"

    fun unescapeString(string: StringLiteralExpression): String = StringBuilder().also {
        string.createLiteralTextEscaper().decode(string.valueRange, it)
    }.toString()
}
